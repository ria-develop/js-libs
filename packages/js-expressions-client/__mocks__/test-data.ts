export const input = {
  user: {
    id: 'uid1',
    firstName: 'John',
    lastName: 'Doe',
    address: 'st. Middle Of Nowhere 4'
  },
  products: [
    [
      {
        name: 'Product 1',
        selected: true
      }
    ],
    [
      {
        name: 'Product 2',
        userComment: 'Need to be reviewed',
        selected: false
      }
    ]
  ],
  options: {
    selectedLocations: ['Location1', 'Location3'],
    deliveryDate: new Date(2021, 1, 2)
  }
};

type InputData = { user; products; options };

const findCompleted = (items) => items.filter((basket) => !basket.some((product) => !product.selected));

const getIfCompleted = (found, numItems) => (found.length === numItems ? found : undefined);

const options = (data) => data.options;

const deliveryDate = (data) => options(data).deliveryDate;

const selectedLocations = (data) => options(data).selectedLocations;

const delivery = (data: InputData): string => deliveryDate(data)?.toISOString();

const locations = (data: InputData): string[] =>
  selectedLocations(data).map((location: string) => (deliveryDate(data) ? location : 'Default'));

const orders = (data: InputData): [] | undefined => getIfCompleted(findCompleted(data.products), data.products.length);

const user = '{user.firstName} {user.lastName}';

const data = [{
  user,
  orders
}, {
  locations
}]

const request = {
  delivery,
  data
}

export const outputShape = {
  request
}
