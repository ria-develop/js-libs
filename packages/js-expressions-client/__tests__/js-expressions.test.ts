import replacer from '../src/js-expressions-client';
import {input, outputShape} from '../__mocks__/test-data';

describe(`given ${JSON.stringify(input)}`, () => {
  it('should match snapshot', () => {
    expect(JSON.stringify(outputShape, replacer(input))).toMatchSnapshot();
  });
});