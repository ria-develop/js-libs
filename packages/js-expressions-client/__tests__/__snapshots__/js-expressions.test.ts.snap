// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given {"user":{"id":"uid1","firstName":"John","lastName":"Doe","address":"st. Middle Of Nowhere 4"},"products":[[{"name":"Product 1","selected":true}],[{"name":"Product 2","userComment":"Need to be reviewed","selected":false}]],"options":{"selectedLocations":["Location1","Location3"],"deliveryDate":"2021-02-01T23:00:00.000Z"}} should match snapshot 1`] = `"{\\"request\\":{\\"delivery\\":\\"2021-02-01T23:00:00.000Z\\",\\"data\\":[{\\"user\\":\\"John Doe\\"},{\\"locations\\":[\\"Location1\\",\\"Location3\\"]}]}}"`;
